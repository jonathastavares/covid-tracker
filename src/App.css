.links {
  color: black;
  text-decoration: none;
}

.link-container {
  margin: 0;
}

.load-container {
  margin: auto 0;
}

.load-title {
  color: #96d6d0;
}

.container {
  position: relative;
  width: 50%;
  margin: 280px auto;
  padding: 20px 40px;
  border-radius: 4px;
  box-sizing: border-box;
  background: #fff;
}

.Loading {
  position: relative;
  display: inline-block;
  width: 100%;
  height: 10px;
  background: #f1f1f1;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  overflow: hidden;
}

.Loading::after {
  content: '';
  position: absolute;
  left: 0;
  width: 0;
  height: 100%;
  border-radius: 4px;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.2);
  animation: load 5s infinite;
}

@keyframes load {
  0% {
    width: 0;
    background: #54bbb4;
  }

  25% {
    width: 40%;
    background: #4bb0a9;
  }

  50% {
    width: 60%;
    background: #3fa59e;
  }

  75% {
    width: 75%;
    background: #329b93;
  }

  100% {
    width: 100%;
    background: #239089;
  }
}
