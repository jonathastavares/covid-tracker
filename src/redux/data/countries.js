import axios from 'axios';

const GET_DATA_SUCCESS = 'covid-tracker/data/GET_DATA_SUCCESS';
const GET_DATA_FAILURE = 'covid-tracker/data/GET_DATA_FAILURE';
const baseURL = 'https://api.covid19tracking.narrativa.com';
const countriesURL = `${baseURL}/api/countries`;
const initialState = [];

export const getData = () => (dispatch) => axios.get(countriesURL).then(
  (response) => {
    const countries = Object.entries(response);
    const newCountries = [];
    for (let i = 0; i < countries[0][1].countries.length; i += 1) {
      const country = {
        name: countries[0][1].countries[i].name,
        link: `${baseURL}${countries[0][1].countries[i].links[0].href}`,
      };
      newCountries.push(country);
    }
    dispatch({ type: GET_DATA_SUCCESS, payload: newCountries });
  },
  (err) => dispatch({ type: GET_DATA_FAILURE, err }),
);

const countriesReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DATA_SUCCESS:
      return [
        ...action.payload,
      ];

    default:
      return state;
  }
};

export default countriesReducer;
