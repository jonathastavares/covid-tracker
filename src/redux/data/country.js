import axios from 'axios';

const GET_DATA_SUCCESS = 'covid-tracker/data/GET_DATA_SUCCESS';
const initialState = [];

export const getData = (countryURL) => (dispatch) => axios.get(countryURL).then(
  (response) => {
    const res = Object.entries(response);
    const payload = res[0][1].dates;
    const country = Object.values(payload)[0].countries;
    const { regions } = Object.values(country)[0];
    for (let i = 0; i < regions.length; i += 1) {
      regions[i].id = i;
    }
    dispatch({ type: GET_DATA_SUCCESS, country: [country], regions });
  },
);

const countryReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DATA_SUCCESS:
      return [
        ...action.country,
        ...action.regions,
      ];
    default:
      return state;
  }
};

export default countryReducer;
