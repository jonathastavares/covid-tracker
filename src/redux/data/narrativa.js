import axios from 'axios';

const GET_DATA_SUCCESS = 'covid-tracker/data/GET_DATA_SUCCESS';
const GET_DATA_FAILURE = 'covid-tracker/data/GET_DATA_FAILURE';

const initialState = [];

export const getData = () => (dispatch) => axios.get('https://api.covid19tracking.narrativa.com/api/countries').then(
  (response) => {
    const countries = Object.entries(response);
    const newCountries = [];
    for (let i = 0; i < countries[0][1].countries.length; i += 1) {
      const country = {
        name: countries[0][1].countries[i].name,
        link: countries[0][1].countries[i].links[0].href,
      };
      newCountries.push(country);
    }
    dispatch({ type: GET_DATA_SUCCESS, payload: newCountries });
  },
  (err) => dispatch({ type: GET_DATA_FAILURE, err }),
);

const narrativaReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DATA_SUCCESS:
      return [
        ...action.payload,
      ];

    default:
      return state;
  }
};

export default narrativaReducer;
