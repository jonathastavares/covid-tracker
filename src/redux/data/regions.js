import axios from 'axios';

const GET_REGIONS_SUCCESS = 'covid-tracker/data/GET_DATA_SUCCESS';
const GET_REGIONS_FAILURE = 'covid-tracker/data/GET_DATA_FAILURE';
const initialState = [];

export const getRegions = (link) => (dispatch) => axios.get(link).then(
  (response) => {
    const regions = Object.entries(response);
    dispatch({ type: GET_REGIONS_SUCCESS, payload: regions });
  },
  (err) => dispatch({ type: GET_REGIONS_FAILURE, err }),
);

const regionsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_REGIONS_SUCCESS:
      return [
        ...action.payload,
      ];

    default:
      return state;
  }
};

export default regionsReducer;
