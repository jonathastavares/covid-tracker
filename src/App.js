import React, { useEffect } from 'react';
import './App.css';
import { useDispatch } from 'react-redux';
import {
  BrowserRouter as Router,
  Route,
  Routes,
  NavLink,
} from 'react-router-dom';
import { getData } from './redux/data/country';
import Country from './components/country';
import Regions from './components/regions';
import Region from './components/region';

const baseURL = 'https://api.covid19tracking.narrativa.com';

function App() {
  const dispatch = useDispatch();
  const today = new Date();
  const year = today.getFullYear();
  const month = () => {
    if ((today.getMonth() + 1) < 10) {
      return `0${(today.getMonth() + 1)}`;
    }
    return (today.getMonth() + 1);
  };
  const day = () => {
    if (today.getDate() < 10) {
      return `0${today.getDate()}`;
    }
    return today.getDate();
  };
  const date = `${year}-${month()}-${day()}`;
  const link = `${baseURL}/api/${date}/country/brazil`;

  useEffect(() => {
    dispatch(getData(link));
  }, []);

  return (
    <Router>
      <nav className="d-flex">
        <div>
          <h1>Covid Tracker</h1>
        </div>
        <ul>
          <li>
            <NavLink
              exact="true"
              to="/"
              className="links"
              activeclassname="active"
            >
              In√≠cio
            </NavLink>
          </li>
        </ul>
      </nav>
      <Routes>
        <Route path="/" element={<Country />} />
        <Route path="regions" element={<Regions />} />
        <Route path="/region" element={<Region />} />
      </Routes>
    </Router>
  );
}

export default App;
