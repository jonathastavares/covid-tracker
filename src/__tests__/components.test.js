import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../App';
import Region from '../components/region';
import data from '../mocks/data';

describe('Testing Components Render and User Interaction: ', () => {
  const region = {
    state: {
      region: {
        name: 'Rio de Janeiro',
        today_confirmed: 521346858,
        today_new_confirmed: 50,
        today_new_deaths: 5,
        source: 'Ministério da Saúde do Brasil',
      },
    },
  };
  const mockStore = configureMockStore([thunk]);
  const store = mockStore(data);

  it('Tests App render', () => {
    const tree = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Tests Region render', () => {
    const tree = render(
      <BrowserRouter>
        <Region key={region.name} region={region.state.region} />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
