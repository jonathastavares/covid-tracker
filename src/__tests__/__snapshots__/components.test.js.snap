// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Render and User Interaction:  Tests App render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container load-container"
      >
        <h1
          class="fs-3 fw-bold text-center load-title"
        >
          Loading content, please wait!
        </h1>
        <div
          class="Loading"
        >
           
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container load-container"
    >
      <h1
        class="fs-3 fw-bold text-center load-title"
      >
        Loading content, please wait!
      </h1>
      <div
        class="Loading"
      >
         
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components Render and User Interaction:  Tests Region render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-6 col-xl-3 text-center card-dark card"
        id="card"
      >
        <a
          class="links align-middle"
          href="/details"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold text-white card-title h5"
            >
              Rio de Janeiro
            </div>
            <div
              class="mb-2 text-white card-subtitle h6"
            >
              Total cases:
              <span
                class="fw-bold"
              >
                521346858
              </span>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-6 col-xl-3 text-center card-dark card"
      id="card"
    >
      <a
        class="links align-middle"
        href="/details"
      >
        <div
          class="card-body"
        >
          <div
            class="fw-bold text-white card-title h5"
          >
            Rio de Janeiro
          </div>
          <div
            class="mb-2 text-white card-subtitle h6"
          >
            Total cases:
            <span
              class="fw-bold"
            >
              521346858
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
